src/simulator_core/infrastructure/plotting.py

```diff
@@ -0,0 +1,112 @@
+#  Copyright (c) 2023. Deltares & TNO
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""Plot result on the map for debugging."""
+import esdl
+import plotly.express as px
+import geopandas as gpd
+from pandas import DataFrame
+import numpy as np
+from shapely.geometry import LineString
+from numpy.typing import NDArray
+from simulator_core.entities.esdl_object import EsdlObject
+
+
+class Plotting:
+    """Plotting plot the results of asset (currently pipe) for inspection."""
+
+    def __init__(self, esdl_object: EsdlObject) -> None:
+        """Initialize plotting instance.
+
+        :param EsdlObject esdl_object: esdl object of the network
+        """
+        self.esdl_object = esdl_object
+
+    def plot_map(self, result: DataFrame) -> None:
+        """Static function to plot the simulation results to the map.
+
+        :param DataFrame result: simulation result in dataframe format
+        """
+        geo_df = self.prepare_data(result)
+        Plotting.plot_line_mapbox(geo_df)
+
+    def prepare_data(self, result: DataFrame) -> gpd.GeoDataFrame:
+        """Function to prepare geopandas dataframe."""
+        geo_df = gpd.GeoDataFrame(
+            columns=['asset_id', 'asset_name', 'geometry', 'mass_flow', 'pressure_in',
+                     'pressure_out', 'temperature_in', 'temperature_out'], crs="EPSG:4326")
+
+        assets = self.get_assets_by_asset_type(esdl.Pipe)
+
+        for asset in assets:
+            if isinstance(asset.geometry, esdl.esdl.Line):
+                coor_pipe = []
+                for point in asset.geometry.point:
+                    coor_pipe.append([point.lon, point.lat])
+
+                last_data_index = len(result) - 1
```

Comment:
Why do you need this index? 

My interpretation:
- You plot for each pipe in the ESDL object the last result. 
- Maybe give the time index of the results to plot as a requirement? Or otherwise, use the dataframe associated with a single time step?

Reply:
yes currently i just take the last value